File: 121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        const int n = prices.size();
        if (n == 0) {
            return 0;
        }
        int first, second, maximum;
        first=second=maximum = 0;
        
        for(int i=0; i< n; i++) {
            if (prices[i] <= prices[first]) {
                first = second= i;
            }
            if (prices[i] > prices[second]) {
                second = i;
                maximum = max(maximum, prices[second] - prices[first]);
            }
        }
        return maximum;
        
    }
};

File: 121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        ans = 0
        minSoFar = prices[0]
        for i in range(1, n):
            ans = max(ans, prices[i] - minSoFar)
            minSoFar = min(minSoFar, prices[i])
        return ans
